lib/board.rb

```diff
@@ -0,0 +1,59 @@
+class Board
+    attr_accessor :cells
+
+    def initialize
+        self.cells = Array.new
+        reset!
+    end
+
+    def reset!
+        (0..8).each {|n| cells[n] = ' '}
+    end
+
+    def display
+        display_row 0
+        puts "-----------"
+        display_row 3
+        puts "-----------"
+        display_row 6
+    end
+
+    def display_row start_cell
+        #avoids having to type in this interpolation monstrosity for each row
+        puts " #{cells[start_cell]} | #{cells[start_cell + 1]} | #{cells[start_cell + 2]} "
+    end
+
+    def position cell_num
+       #need to convert to integer and correct for humans thinking 1-9, computer 0-8
+       idx = cell_num.to_i - 1
+       cells[idx]
+    end
+
+    def full?
+        !cells.detect{|pos| pos == " "}
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.